quantum/quantum.c

```diff
@@ -214,6 +214,13 @@ bool process_record_quantum(keyrecord_t *record) {
 
 #if defined(SECURE_ENABLE)
     if (!preprocess_secure(keycode, record)) {
+        // If keys are being held when this is triggered, they may not be released properly
+        // this can result in stuck keys, mods and layers.  To prevent that, manually
+        // clear these, when it is triggered.
+        if (!record->event.pressed) {
```

Comment:
This happens on every keypress, rather than the transition from inscure -> unlocking.

I somewhat prefer the addition of a "request unlock" keycode, or just letting the caller handle the case if they manually call `secure_request_unlock`.

Reply:
Can do that.